src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -94,50 +103,61 @@ public static AnalysisConfig GenerateFile(ProcessedArgs localSettings,
+    private static void HandleCoverageExclusions(AnalysisConfig config, ProcessedArgs localSettings, IDictionary<string, string> serverProperties)
     {
-        HashSet<string> coveragePropertyNames =
-        [
-            "sonar.cs.vscoveragexml.reportsPaths",
-            "sonar.cs.dotcover.reportsPaths",
-            "sonar.cs.opencover.reportsPaths",
-        ];
-        var allProperties = localSettings.CmdLineProperties.GetAllProperties().ToList();
-        var coveragePaths = string.Join(",", allProperties.Where(x => coveragePropertyNames.Contains(x.Id)).Select(x => x.Value));
-        if (!localSettings.ScanAllAnalysis      // if scanAll analysis is disabled, we will not pick up the coverage files anyways
-            || coveragePaths.Length == 0)       // if there are no coverage files, there is nothing to exclude
+        var localProperties = localSettings.AllProperties().ToList();
+        var localCoveragePaths = string.Join(",", localProperties.Where(x => CoveragePropertyNames.Contains(x.Id)).Select(x => x.Value));
+        var serverCoveragePaths = string.Join(",", serverProperties.Where(x => CoveragePropertyNames.Contains(x.Key)).Select(x => x.Value));
+        if (!localSettings.ScanAllAnalysis                                  // If scanAll analysis is disabled, we will not pick up the coverage files anyways
+            || localCoveragePaths.Length + serverCoveragePaths.Length == 0) // If there are no coverage files, there is nothing to exclude
         {
-            return allProperties;
+            return;
         }
+        var localExclusions = localSettings.GetSetting(SonarExclusions, string.Empty);
+        var serverExclusions = serverProperties.ContainsKey(SonarExclusions) ? serverProperties[SonarExclusions] : string.Empty;
+        AddCoverageExclusions(localCoveragePaths.Length > 0 ? localCoveragePaths : serverCoveragePaths);
 
-        if (allProperties.Find(x => x.Id == "sonar.exclusions") is { } exclusionsProperty)
+        void AddCoverageExclusions(string coveragePaths)
         {
-            exclusionsProperty.Value = exclusionsProperty.Value + "," + coveragePaths;
+            if (string.IsNullOrEmpty(localExclusions) && string.IsNullOrEmpty(serverExclusions))
+            {
+                localExclusions = coveragePaths;
+            }
+            else if (string.IsNullOrEmpty(localExclusions))
+            {
+                localExclusions = string.Join(",", serverExclusions, coveragePaths);
+            }
+            else
+            {
+                localExclusions += "," + coveragePaths;
+            }
+            // Recreate LocalSettings property
+            if (config.LocalSettings.Exists(x => x.Id == SonarExclusions)
+                || !string.IsNullOrWhiteSpace(localExclusions))
+            {
+                config.LocalSettings.RemoveAll(x => x.Id == SonarExclusions);
+                AddSetting(config.LocalSettings, SonarExclusions, localExclusions);
```

Comment:
Very optional, but if you want to be spicy you can merge the exists and the remove like so:

```csharp
            if (config.LocalSettings.RemoveAll(x => x.Id == SonarExclusions) > 0
                || !string.IsNullOrWhiteSpace(localExclusions))
            {
                AddSetting(config.LocalSettings, SonarExclusions, localExclusions);
            }
```

Reply:
I would prefer to keep it this way for readability purposes, but that's really nice!